### PR TITLE
[dev inspect] Move call entry point

### PR DESCRIPTION
**Pull request by tnowacki**
- Add direct Move call entry point for dev inspect

### CODE REVIEW

File: crates/sui-core/src/authority.rs [hunk 1]
```diff
@@ -1005,6 +1005,46 @@ impl AuthorityState {
...
+                sender,
+                move_call,
+                transaction_digest,
+                transaction_dependencies,
+                &self.move_vm,
+                gas_status,
+                self.epoch(),
+            );
+
+        DevInspectResults::new(effects, execution_result, self.module_cache.as_ref())
```
Comment (gegaowp) [inline crates/sui-core/src/authority.rs:1045]: wondering are `effects` required here, iiuc `effects` are about txns and move-call is only about `execution_result` ?
if we only need `execution_result`, is it possible to get rid of txn_digest?

Comment (tnowacki) [reply crates/sui-core/src/authority.rs:1046]: No, txn digest is used throughout the system. And the effects gives you information about objects created/modified/deleted.

File: crates/sui-core/src/execution_engine.rs [hunk 2]
```diff
@@ -640,6 +640,65 @@ fn transfer_sui<S>(
     Ok(())
 }
 
+pub(crate) fn manual_execute_move_call<
```
Comment (amnn) [inline crates/sui-core/src/execution_engine.rs:643]: I kind of expected this function or one like it to be used in both the dry-run path and the normal move-call path to keep the behaviour between those two in sync as much as possible.  I don't think that's what's happening here, but is that possible?

Comment (tnowacki) [reply crates/sui-core/src/execution_engine.rs:643]: This is a helper function for dev_inspect (and  maybe later dry_run) for manually invoking a Move function without having to have a gas object. So the usage will be rather niche (and a tad unsafe)

**Review by amnn — APPROVED** [review]
LGTM!

**Comment by amnn** [comment]
But will leave it with you all if you want to update to add the digest generation in the authority.

**Comment by tnowacki** [comment]
> But will leave it with you all if you want to update to add the digest generation in the authority.

I think we should do it for all the end points, but I'll do it just the dev_inspect_move_call for now

**Comment by amnn** [comment]
All as in all the dry run end points, or all as in every endpoint?

**Comment by tnowacki** [comment]
> All as in all the dry run end points, or all as in every endpoint?

The dry run and dev inspect endpoints. The normal endpoints already have the  transaction digest hashed as a part of the signed certificate

**Pull request merged**